### ISSUE TITLE
[Bug] Left in an empty server when it is shutdown while you are in the bigmap

### ISSUE THREAD

**Issue by NicholasMcMillan03** [Starystars67/BeamNG-MP#363]
**Have you read our FAQ/Wiki/#before-you-ask ?**
Yes

**Describe the bug**
If you have the bigmap open while the server you are in shuts down you will not get the shutdown message and when you close bigmap you will be in an empty server with ? players (basically singleplayer not sure if there is a name for that sate)

**To Reproduce**
1. Join server
3. open bigmap
4. shutdown server
5. close bigmap

**Expected behavior**
When you close bigmap you should be left with the server shut down message, or you should have bigmap closed on you and be shown the message if that is possible, either seems like acceptable behavior to me.

**Screenshots**
![image](https://github.com/BeamMP/BeamMP/assets/57968816/2752e077-ae87-4327-986f-265f978730b1)
what you see after exiting bigmap when server has been shutdown while you were in it

**Desktop (please complete the following information):**
 - OS + Version: Ubuntu Desktop 22.04.2 LTS
 - Browser (If applicable): N/A

**Additional context**
N/A

**Comment by Starystars67** [comment]
Hi there, Thank you for raising this issue.

This is an interesting one and will require some digging into.

Also thank you for being thorough in providing your information.

Kind regards,
Titch

**Comment by Starystars67** [comment]
This issue looks to be related to the fact that the no relevant BeamMP UI modules are loaded during the use of the "bigmap" and as such there is no prompt to display. The event is then forgotten about and that is the end of handling.

This needs to be fixed, likely by moving the prompt code to the main / top level files rather than a sub module.

**Comment by O1LER** [comment]
You can get rid of the server shutdown message, or any other similar one, by just refreshing the UI, F5. So you can "stay" in an empty server not just by using the bigmap.

So i guess leaving bigmap might just do that, refreshing the UI

**Comment by Starystars67** [comment]
> You can get rid of the server shutdown message, or any other similar one, by just refreshing the UI, F5. So you can "stay" in an empty server not just by using the bigmap.
> 
> So i guess leaving bigmap might just do that, refreshing the UI

The issue here was that you received no notification that the server had shutdown or kicked  you while you are in the bigmap view.

**Comment by Starystars67** [comment]
I have been doing a little work towards this issue, Here is the result of the work. 
I am looking for some input to shape this ideally please.

<img width="1920" alt="image" src="https://github.com/BeamMP/BeamMP/assets/15389482/7126c1b3-8b30-4d8c-a72b-acff06ad26b0">

Also it fully supports translations too which is nice!

**Comment by O1LER** [comment]
But is that actually the case? Maybe the bigmap is just placed infront of the message, and when exiting it refreshes the UI, removing the message

**Comment by O1LER** [comment]
> I have been doing a little work towards this issue, Here is the result of the work. I am looking for some input to shape this ideally please.
> 
> <img alt="image" width="1920" src="https://private-user-images.githubusercontent.com/15389482/289076462-7126c1b3-8b30-4d8c-a72b-acff06ad26b0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIwNDE2MTgsIm5iZiI6MTcwMjA0MTMxOCwicGF0aCI6Ii8xNTM4OTQ4Mi8yODkwNzY0NjItNzEyNmMxYjMtOGIzMC00ZDhjLWE3MmItYWNmZjA2YWQyNmIwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA4VDEzMTUxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk5MWUxNzFhM2JjYjIxMjk2MDY5ODE4NjVmZTdlMGE2ODY3ZmI0NWZjMWMzNzgxMzU0MzkyYWRjZDU5ZWI5ODcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.t1kwTRNwmTPsZwAgugzYhF42A-9mkYt1UOnlzvUP8G8">
> Also it fully supports translations too which is nice!

I see, nvm then :-)

**Comment by Starystars67** [comment]
> But is that actually the case? Maybe the bigmap is just placed infront of the message, and when exiting it refreshes the UI, removing the message

Yes sadly it is, The issue is that the UI system that renders our normal quit/disconnect message is handled by the session UI app which is technically unloaded when you enter the big map, as such we have no control over it and the UI element is therefore not rendered.

The approach I have taken here uses the root level UI system of the game which means it comes before all other UI screens and as such is able to be rendered at any time including above other aspects or even during the loading into a server.
